### DISCUSSION TITLE
how to deploy sveltekit on digitalocean

### DISCUSSION THREAD

**Discussion by applemate** [sveltejs/kit#10396] [Q&A]
### Describe the problem

how to deploy sveltekit on digitalocean. I there a guideline how to do this ? 

### Describe the proposed solution

I got error "vite not found" during the deploy process. 

### Importance

i cannot use SvelteKit without it

**Comment by ghostdevv** [comment]
I'm not too familiar with Digital Ocean but I would imagine you would use the node adapter? Though it depends on what service you are using from them

**Reply by applemate** [reply]
I'm not using adapter. 

Digital ocean app platform is the service I use

**Reply by ghostdevv** [reply]
A quick google found [this article](https://linu.us/deploy-a-sveltekit-app-to-digitalocean), maybe that will help?

**Reply by applemate** [reply]
that link is what's I follow from the start. it doesn't work anymore. need update.

**Comment by anasfik** [comment]
Thank you so much.. this worked and I deployed my svelte kit app like a charm on DO!

**Reply by applemate** [reply]
can you share how you do it ?

**Reply by anasfik** [reply]
Yes sure.

1. After I created my svelte kit app which ran normally on my local host.

2. Created a new app from the Digital Ocean platform, I chose GitHub to host my repository.
![image](https://github.com/sveltejs/kit/assets/25140579/0a9bfd81-64c9-440b-b4be-a85a1b4bfa86)

3. In the settings of the Digital Ocean app dashboard, I had to set the build & run commands in addition to the port configs to:

- Build: `npm run build`
- host: `npm run preview`
- Port: you will need to set this to the port you saw in your localhost

4. Select other options such as your plan and environment variables...

5. Then review and hit the deploy button.

Which successfully built the app, but failed in deployment. I didn't know what was wrong until I learned that I should expose the host of the app by following these steps:

1. In my app repository, I opened `package.json`.

2. In the `preview` script command, I had to add the `--host` flag to it's value, so it is like this:
![image](https://github.com/sveltejs/kit/assets/25140579/65c6eaac-4d58-4ff9-b7d6-fe7f368fda7f)

3. Then I committed my changes and pushed them to Github.

4. Then I followed the same first steps above, and it got deployed successfully.

My app is deployed now at [https://langsync.app](https://langsync.app)

**Reply by david-plugge** [reply]
I'm sure that the preview script isn't the way you should run your server in production.
Setup the node adapter (https://kit.svelte.dev/docs/adapter-node) and change your host script from `npm run preview` to `node build`.

The port will also change to 3000.

**Reply by applemate** [reply]
@david-plugge can you share details how you do it ?

**Reply by david-plugge** [reply]
I don't use digital ocean but if it works with the preview script the steps I outlined above should be enough.

**Comment by applemate** [comment]
I can do it now, just follow the steps here: https://kit.svelte.dev/docs/adapter-node

a document by sveltekit team

**Reply by kelsiesmurphy** [reply]
Could you share further details on this Craig? It would be great to see your svelte.config.js, and package.json file, and what your settings were in Digital Ocean to get this running. Thanks!

**Comment by dmdin** [comment]
The question is more related to DevOps practices. The process is the same for each cloud provider if you are using Virtual Machine. Usually it's made with a Dockerfile for your project, to be easy to run inside any environment. If you have a static site with prerendered pages, you don't need adapter-node

**Comment by binhkid2** [comment]
Add this to package.json
  "engines": {
    "node": "18.20.4"
  },
  and "preview": "vite preview --host ",
  use @sveltejs/adapter-node"
  and use NPM, PNPM or Yarn not bun or something else
  When deploy change build command is npm run preview  and PORT http is 4173
  This work for me 👍

**Comment by benmccann** [comment]
You can use adapter-static with the digital ocean static site component. Or you can use adapter-node with the digital ocean web service component.
